cmake/python.cmake

```diff
@@ -55,8 +55,17 @@ if(PYTHONLIBS_FOUND AND PYTHON_MODULE_EXTENSION)
     return()
 endif()
 
+# On Mac, with Maya 2022+, the Python binaries link against a non-existent path
+# maya2022/Maya.app/Contents/Frameworks/Python.framework/Versions/Current/bin/python -> /Library/Frameworks/Python.framework/Versions/3.7/Python
+# However, just using the mayapy executable is good enough for the build system
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+
+    MESSAGE(STATUS "Setting Python_EXECUTABLE to MayaPy ${MAYA_PY_EXECUTABLE}")
+    set(Python_EXECUTABLE ${MAYA_PY_EXECUTABLE})
+    set(Python_Interpreter_FOUND TRUE)
+
 # Use the Python module to find the python lib.
-if(BUILD_WITH_PYTHON_3)
+elseif(BUILD_WITH_PYTHON_3)
```

Comment:
I understand what you are doing here, but I'm a little wary about the _elseif_ here which means that it now skips the find_package(Python). Above you are only setting the Python_EXECUTABLE var. Does the find_package(Python) set other stuff that will be skipped now?

Reply:
From my searching through the current CMake files, I couldn't find anything else that gets used from the standard FindPython.cmake 

However I do see that if I only set the Python_EXECUTABLE and not make it supersede the find_package calls, it'll still work. So I've updated the PR with that, so it's less invasive, in case anyone else adds stuff in the future that find_package depends on

Reply:
Great thanks.